clearnode/app_session_service.go

```diff
@@ -18,6 +18,66 @@ const (
 	ErrRecordTransaction  = "failed to record transaction"
 )
 
+// SessionKeyContext holds information about session key usage.
+type SessionKeyContext struct {
+	SessionKeyAddress *string
+	IsDirectSig       bool
+	HasSignature      bool
```

Comment:
Aren't `SessionKeyAddress` and `IsDirectSig` duplicating each other? I mean, if `SessionKeyAddress == nil`, it means that `IsDirectSig == true` and vice versa for the opposite equality signs.